interfaces/mount/spec.go

```diff
@@ -42,6 +44,16 @@ func (spec *Specification) AddMountEntry(e Entry) error {
 func (spec *Specification) MountEntries() []Entry {
 	result := make([]Entry, len(spec.mountEntries))
 	copy(result, spec.mountEntries)
+	// Number each entry, in case we get clashes this will automatically give
+	// them unique names.
+	count := make(map[string]int, len(spec.mountEntries))
```

Comment:
This is nice - but I slightly wonder about the results here. In the corresponding test I see `/var/snap/consumer/common/import/directory` and `/var/snap/consumer/common/import/directory-2` one is ro, one is rw. This makes me wonder two things: a) is the order stable, i.e. is directory-2 always the rw one? b) will that be useful for an application?  i.e. what is the advantage over just rejecting clashing names?

Reply:
I'll answer the ro/rw aspect separately.

The de-clashing allows us to have a system where plugin authors don't have to coordinate with each on directory names. The de-clashing aspect is also described on the forum (link is in the commit message).

Reply:
Could we log a message when we need to declash?